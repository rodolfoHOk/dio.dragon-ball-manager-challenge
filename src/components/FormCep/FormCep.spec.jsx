import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import FormCep from './FormCep';
import { act } from 'react-dom/test-utils';

global.fetch = jest.fn().mockImplementation(() =>
  Promise.resolve({
    ok: true,
    status: 200,
    json: async () => ({
      cep: '01001000',
      bairro: 'Sé',
      logradouro: 'Praça da Sé',
    }),
  })
);

it('should render address form', async () => {
  const { container, getByPlaceholderText } = render(<FormCep />);

  const cepInput = getByPlaceholderText('CEP');
  fireEvent.change(cepInput, { target: { value: '01001000' } });

  await act(() => global.fetch);

  expect(global.fetch).toHaveBeenCalledTimes(1);
  expect(global.fetch).toHaveBeenCalledWith(
    'https://viacep.com.br/ws/01001000/json/'
  );
  expect(container).toMatchSnapshot();
});
