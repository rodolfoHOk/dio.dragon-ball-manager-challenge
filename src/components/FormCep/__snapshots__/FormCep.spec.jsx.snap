// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render address form 1`] = `
<div>
  <form
    class="css-1ldrcmr"
  >
    <div
      class="css-1viozcy"
    >
      <div
        class="css-1e8zq2i"
      >
        <label
          class="css-knfhfe"
          for="cep"
        >
          Cep
        </label>
        <input
          class="css-4y926w"
          id="cep"
          name="cep"
          placeholder="CEP"
          type="text"
          value="01001000"
        />
      </div>
      <div
        class="css-1e8zq2i"
      >
        <label
          class="css-knfhfe"
          for="bairro"
        >
          Bairro
        </label>
        <input
          class="css-4y926w"
          id="bairro"
          name="bairro"
          placeholder="bairro"
          type="text"
          value="Sé"
        />
      </div>
      <div
        class="css-1e8zq2i"
      >
        <label
          class="css-knfhfe"
          for="logradouro"
        >
          Logradouro
        </label>
        <input
          class="css-4y926w"
          id="logradouro"
          name="logradouro"
          placeholder="logradouro"
          type="text"
          value="Praça da Sé"
        />
      </div>
    </div>
  </form>
</div>
`;
